### ISSUE TITLE
db is not a function

### ISSUE THREAD

**Issue by chongwang87** [mongo-express/mongo-express#361]
```
TypeError: db is not a function
    at /usr/local/lib/node_modules/mongo-express/lib/db.js:114:53
    at Array.forEach (<anonymous>)
    at setupAuthDatabases (/usr/local/lib/node_modules/mongo-express/lib/db.js:111:25)
    at processOpenDatabase (/usr/local/lib/node_modules/mongo-express/lib/db.js:158:9)
    at /usr/local/lib/node_modules/mongo-express/node_modules/mongodb/lib/db.js:236:5
    at Server.connectHandler (/usr/local/lib/node_modules/mongo-express/node_modules/mongodb/lib/server.js:337:7)
    at Object.onceWrapper (events.js:316:30)
    at emitOne (events.js:115:13)
    at Server.emit (events.js:210:7)
    at /usr/local/lib/node_modules/mongo-express/node_modules/mongodb-core/lib/topologies/server.js:297:14
/usr/local/lib/node_modules/mongo-express/node_modules/mongodb/lib/server.js:340
      process.nextTick(function() { throw err; })
```

**Comment by karlitos** [comment]
I have same problem. I installed _mongo-expres_ globally under ubuntu and tried to start it with:
`mongo-express --username myUser --password myPassword --database db_name --port 8091`

If I use the same credentials for the __mongo shell__
`mongo -u myUser -p myPassword --db_name db_name` I can log in without any problems.

**Comment by creeperyang** [comment]
Exactly the same problem. And dive into the code:

```js
setupAuthDatabases(db)
// db here is an instance of Db, not a funtion
// SO error will throw when code runs into:
//
// connectionData.connections[auth.database] = db(auth.database);
//
```

I don't why use `db(auth.database)`, maybe just a legacy code?

**Comment by karlitos** [comment]
Hello @creeperyang I tried your fix, but it did not help me any further. Instead of the _db is not a funciton_ error message I get
```javascript
Connecting to development_configuration...
{ MongoError: Authentication failed.
    at Function.MongoError.create (/usr/lib/node_modules/mongo-express/node_modules/mongodb-core/lib/error.js:31:11)
    at /usr/lib/node_modules/mongo-express/node_modules/mongodb-core/lib/connection/pool.js:483:72
    at authenticateStragglers (/usr/lib/node_modules/mongo-express/node_modules/mongodb-core/lib/connection/pool.js:429:16)
    at Connection.messageHandler (/usr/lib/node_modules/mongo-express/node_modules/mongodb-core/lib/connection/pool.js:463:5)
    at Socket.<anonymous> (/usr/lib/node_modules/mongo-express/node_modules/mongodb-core/lib/connection/connection.js:319:22)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    at TCP.onread (net.js:547:20)
  name: 'MongoError',
  message: 'Authentication failed.',
  ok: 0,
  errmsg: 'Authentication failed.',
  code: 18,
  codeName: 'AuthenticationFailed' }
Could not authenticate to database "development_configuration"
```

No problems connecting with the given user, password to the _development_configuration_ database in mongo shell.

**Comment by creeperyang** [comment]
Yeah, just try `mongo-express -a -u user -p pwd ` will work correctly.

By the way, it's ok to print error here because `authenticate()` it self really fails the `auth` request.

**Comment by karlitos** [comment]
By using the mongoDB admin user credentials I was a able to start mongo express with the _-a_ flag. Does this mean I need to use the _admin user_ for running the mongo-express or can I use the regular one I created for reading/writing ?

**Comment by creeperyang** [comment]
@karlitos Not quite sure. Only after looking through more code and I can answer it.  

May it's better to wait for the repo's owner/maintainers to give a detailed explanation.

**Comment by heartnetkung** [comment]
I faced the same issue and I found out that it only happens in the newer version of mongo-express. So, here is the old working image on docker hub in case anyone want a quick fix.

```
https://hub.docker.com/r/heartnetkung/mongo-express/
```

**Comment by dozoisch** [comment]
published v0.42.3 with https://github.com/mongo-express/mongo-express/pull/366 which should fix the issue.
